packages/tangle/messagefactory.go

```diff
@@ -90,6 +90,19 @@ func (f *MessageFactory) IssuePayload(p payload.Payload) (*Message, error) {
 	weakParents := make([]MessageID, 0)
 
 	issuingTime := clock.SyncedTime()
+
+	// due to the ParentAge check we must ensure that we set the right issuing time.
+	if t != nil {
+		for _, parent := range strongParents {
+			t[0].Message(parent).Consume(func(msg *Message) {
+				if msg.ID() != EmptyMessageID && !msg.IssuingTime().Before(issuingTime) {
+					time.Sleep(msg.IssuingTime().Sub(issuingTime) + 1*time.Nanosecond)
+					issuingTime = clock.SyncedTime()
+				}
+			})
+		}
+	}
```

Comment:
This seems more like a quick hack. Not sure if necessary now but there could be a flag if it should wait for the time or otherwise return an error.